servers/visual/visual_server_scene.cpp

```diff
@@ -105,7 +105,12 @@ void VisualServerScene::camera_set_use_vertical_aspect(RID p_camera, bool p_enab
 
 /* SPATIAL PARTITIONING */
 VisualServerScene::SpatialPartitionID VisualServerScene::SpatialPartitioningScene_BVH::create(Instance *p_userdata, const AABB &p_aabb, int p_subindex, bool p_pairable, uint32_t p_pairable_type, uint32_t p_pairable_mask) {
-	return _bvh.create(p_userdata, p_aabb, p_subindex, p_pairable, p_pairable_type, p_pairable_mask) + 1;
+#if defined(DEBUG_ENABLED) && defined(TOOLS_ENABLED)
+	// we are relying on this instance to be valid in order to pass
+	// the visible flag to the bvh.
+	CRASH_COND(!p_userdata);
```

Comment:
`CRASH_COND` is like a reverse assert, it should only be used here if we're sure that this can never happen unless we (engine devs) messed up. If it's a situation which could happen in practice, it would be better to handle it gracefully with an `ERR_PRINT` and defaulting to `visible = true` or something. We don't want the engine to crash if users did something wrong.

Reply:
Yup it's used as a development assert. I'm quite a fan of these, to check code flow conditions and catch problems early caused by code modifications (rather than runtime), but be compiled out in release. We should probably have a macro for this. I'm surprised they are not used more to be honest.